GA/utils.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime
+
+class Utils:
+    # Dictionary mapping slot numbers to time intervals in 12-hour format
+    time_slots = {
+        1: "9:00 - 9:55",
+        2: "9:55 - 10:50",
+        3: "11:10 - 12:05",
+        4: "12:05 - 1:00",
+        5: "1:20 - 2:15",
+        6: "2:15 - 3:10",
+        7: "3:30 - 4:25",
+    }
+
+    @staticmethod
+    def convert_to_12_hour_format(time_24):
+        """Convert 24-hour time format to 12-hour format with AM/PM using datetime."""
+        time_obj = datetime.strptime(time_24, "%H:%M")
```

Comment:
Without AM/PM, strptime defaults to assuming morning times, and comparing times could lead to false matches. For example, "9:00 - 9:55" in time_slots will always be interpreted as morning, even if the intended time was 9:00 PM - 9:55 PM.